site/content/in-dev/unreleased/configuring-polaris-for-production.md

```diff
@@ -23,114 +23,179 @@ type: docs
+the container.
+
+Be sure to secure your metastore backend since it will be storing credentials and catalog metadata.
+
+### Bootstrapping
+
+Before using Polaris, you must **bootstrap** the metastore manager. This is a manual operation that
+must be performed **only once** in order to prepare the metastore manager to integrate with Polaris.
+
+By default, when bootstrapping a new realm, Polaris will create randomised `CLIENT_ID` and
+`CLIENT_SECRET` for the `root` principal and store their hashes in the metastore backend.
+
+Depending on your database, this may not be convenient as the generated credentials are not stored
+in clear text in the database.
+
+In order to provide your own credentials for `root` principal (so you can request tokens via
+`api/catalog/v1/oauth/tokens`), there are two approaches:
+
+1. Use the Polaris Admin Tool to bootstrap the realm and set its `root`
+   principal credentials.
+2. Set the `root` principal credentials when deploying Polaris for the first time.
+
+The first approach is recommended as it is more flexible (you can bootstrap new realms at any time
+without incurring in a Polaris downtime). The second approach is useful for testing and development,
+but can be used in production as well.
+
+In order to bootstrap root credentials for a realm named `my-realm` when deploying Polaris, set the
+following environment variables:
+
+```
+export POLARIS_BOOTSTRAP_CREDENTIALS=my-realm,my-client-id,my-client-secret
 ```
 
+If the realm hasn't been bootstrapped yet, Polaris will create the realm and the `root` principal
+with the provided credentials upon first usage of that realm. If the realm already exists, Polaris
+will not attempt to update the `root` principal credentials.
+
 You can verify the setup by attempting a token issue for the `root` principal:
 
 ```bash
-curl -X POST http://localhost:8181/api/catalog/v1/oauth/tokens -d "grant_type=client_credentials&client_id=my-client-id&client_secret=my-client-secret&scope=PRINCIPAL_ROLE:ALL"
+curl -X POST http://localhost:8181/api/catalog/v1/oauth/tokens \
+  -d "grant_type=client_credentials" \
+  -d "client_id=my-client-id" \
+  -d "client_secret=my-client-secret" \
+  -d "scope=PRINCIPAL_ROLE:ALL"
 ```
 
-which should return:
+Which should return an access token:
 
 ```json
-{"access_token":"...","token_type":"bearer","issued_token_type":"urn:ietf:params:oauth:token-type:access_token","expires_in":3600}
+{
+  "access_token": "...",
+  "token_type": "bearer",
+  "issued_token_type": "urn:ietf:params:oauth:token-type:access_token",
+  "expires_in": 3600
+}
 ```
 
-Note that if you used non-default realm name, for example, `iceberg` instead of `default-realm` in your `polaris-server.yml`, then you should add an appropriate request header:
+Note that if you used a realm name that is not the default realm name, then you should add an
+appropriate request header to the `curl` command, for example:
```

Comment:
Missing of realm header will be resolved to the first realm in the configuration `polaris.realm-context.realms`, right?
```suggestion
appropriate request header to the `curl` command, otherwise the Polaris will resolve the first realm in the configuration `polaris.realm-context.realms`, here is an example to set realm header:
```